core/lib/types/src/api/mod.rs

```diff
@@ -834,6 +834,36 @@ pub struct TransactionExecutionInfo {
     pub execution_info: Value,
 }
 
+/// The fee history type returned from `eth_feeHistory` call.
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FeeHistory {
+    /// Lowest number block of the returned range.
+    pub oldest_block: BlockNumber,
```

Comment:
This wasn't done for existing API types because of legacy reasons, but for new types, would it be possible to embed the `web3::FeeHistory` type instead of copy-pasting it?

```rust
#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FeeHistory {
    #[serde(flatten)]
    pub inner: web3::FeeHistory,
    /// An array of effective pubdata price. Note, that this field is L2-specific and only provided by L2 nodes.
    #[serde(default)]
    pub pubdata_price: Vec<U256>,
}
```

Would make it more difficult to screw up / easier to maintain, and clearly show the inheritance between the types.